client/my-sites/comments/comment-list/index.jsx

```diff
@@ -117,10 +117,12 @@ export class CommentList extends Component {
 
 	isCommentPersisted = commentId => -1 !== this.state.persistedComments.indexOf( commentId );
 
-	// TODO: replace with array logic after persistedComments is merged
-	isCommentSelected = commentId => !! this.state.selectedComments[ commentId ];
+	isCommentSelected = commentId => !! find( this.state.selectedComments, { commentId } );
 
-	isSelectedAll = () => this.getComments().length === size( this.state.selectedComments );
+	isSelectedAll = () => {
+		const visibleComments = this.getCommentsPage( this.getComments(), this.state.page );
+		return visibleComments.length === this.state.selectedComments.length;
```

Comment:
This probably won't happen, but let's also make sure the length is greater than 0 too.